### ISSUE TITLE
Please Help ? No SOURCES given to target: FSPT_MSG

### ISSUE THREAD

**Issue by muratoznn** [chunyeow/openairinterface5g#9]
Why I am Taking This Error ? You Know ?

murat@murat-VivoBook-14-ASUS-Laptop-X407MA-X407MA:~/openairinterface5g/cmake_targets$ sudo ./build_oai  --eNB -x -w LMSSDR
[sudo] password for murat: 
Will compile eNB
Will generate the software oscilloscope features
Setting hardware to: OAI_LMSSDR
CMAKE_CMD=cmake ..
No transport protocol has been selected (TP set to None)
RF HW set to OAI_LMSSDR
Flags for Deadline scheduler: False
Flags for CPU Affinity: False
2. Setting the OAI PATHS ...
OPENAIR_DIR    = /home/murat/openairinterface5g
3. building the compilation directives ...
CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMAKE_BUILD_TYPE is RelWithDebInfo
Architecture is x86_64
C_FLAGS_PROCESSOR is   -msse4.1 -mssse3 
git found: /usr/bin/git
calling ASN1C_PREFIX=F1AP_ asn1c -gen-PER -fcompound-names -no-gen-example -findirect-choice -fno-include-deps -D /home/murat/openairinterface5g/cmake_targets/lte_build_oai/build/CMakeFiles/F1AP_R15.2.1 /home/murat/openairinterface5g/openair2/F1AP/MESSAGES/ASN1/R15.2.1/F1AP-CommonDataTypes.asn;/home/murat/openairinterface5g/openair2/F1AP/MESSAGES/ASN1/R15.2.1/F1AP-Constants.asn;/home/murat/openairinterface5g/openair2/F1AP/MESSAGES/ASN1/R15.2.1/F1AP-PDU-Descriptions.asn;/home/murat/openairinterface5g/openair2/F1AP/MESSAGES/ASN1/R15.2.1/F1AP-PDU-Contents.asn;/home/murat/openairinterface5g/openair2/F1AP/MESSAGES/ASN1/R15.2.1/F1AP-IEs.asn;/home/murat/openairinterface5g/openair2/F1AP/MESSAGES/ASN1/R15.2.1/F1AP-Containers.asn
DEADLINE_SCHEDULER flag  is False
CPU_Affinity flag is False
/home/murat/openairinterface5g/cmake_targets/tools/generate_protobuf: satır 14: protoc-c: komut yok
calling protoc_call=/home/murat/openairinterface5g/cmake_targets/tools/generate_protobuf FSPT_C_DIR=/home/murat/openairinterface5g/cmake_targets/lte_build_oai/build/CMakeFiles/FSPT_V2 FSPT_MSG_DIR=/home/murat/openairinterface5g/targets/COMMON/MESSAGES/V2 FSPT_MSG_FILES=/home/murat/openairinterface5g/targets/COMMON/MESSAGES/V2/flexsplit.proto
/home/murat/openairinterface5g/cmake_targets/tools/generate_protobuf: satır 14: protoc-c: komut yok
fspt c dir is : /home/murat/openairinterface5g/cmake_targets/lte_build_oai/build/CMakeFiles/FSPT_V2
-- Checking for one of the modules 'openpgm-5.1;openpgm-5.2'
NETTLE VERSION_INSTALLED  = 3.2
NETTLE_VERSION_MAJOR = 3
NETTLE_VERSION_MINOR = 2
-- Configuring done
CMake Error at /home/murat/openairinterface5g/cmake_targets/CMakeLists.txt:1080 (add_library):
  Cannot find source file:

    /home/murat/openairinterface5g/cmake_targets/lte_build_oai/build/CMakeFiles/FSPT_V2/flexsplit.pb-c.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
Call Stack (most recent call first):
  CMakeLists.txt:19 (include)

CMake Error at /home/murat/openairinterface5g/cmake_targets/CMakeLists.txt:1005 (add_library):
  Cannot find source file:

    /home/murat/openairinterface5g/cmake_targets/lte_build_oai/build/CMakeFiles/FLPT_V2/header.pb-c.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
Call Stack (most recent call first):
  CMakeLists.txt:19 (include)

CMake Error at /home/murat/openairinterface5g/cmake_targets/CMakeLists.txt:1080 (add_library):
  No SOURCES given to target: FSPT_MSG
Call Stack (most recent call first):
  CMakeLists.txt:19 (include)

CMake Error at /home/murat/openairinterface5g/cmake_targets/CMakeLists.txt:1005 (add_library):
  No SOURCES given to target: FLPT_MSG
Call Stack (most recent call first):
  CMakeLists.txt:19 (include)

CMake Generate step failed.  Build files cannot be regenerated correctly.
build have failed

**Comment by davmar88** [comment]
Did you install Google Protobuf and proto-c?

**Comment by chunyeow** [comment]
./cmake_targets/build_oai -I 	
to install the necessary packages first

**Comment by davmar88** [comment]
Hi Chunyeow, yes, that is correct, but personally I struggled with the -I argument because it would fail due to certificate issues, thus I had to keep install manually and manually look what it needed, and I see that this error, which I also had, is because the compiler cannot find the correct language packs to translate the extensions. Installing Protobuf, proto2/proto3 and protoc fixed my issues. Also if you installed these packages I mentioned in the above line, and it gives a new error where it cannot find certain files, you go to that specific folder and you create those empty files with sudo nano so that the compiler can find them and move on building the solution.
